### PR TITLE
Lunar + Lutris PR

### PR DESCRIPTION
**Pull request by LogicApples**
Adds support for Lunar Client and Epic Games through Lutris in Linux.

### CODE REVIEW

**Comment by LogicApples** [comment]
this should be working now. Test it on another linux machine first so we can be sure.
If there is anything wrong with the pr hit me up on Discord.

Goodnight!

**Comment by DashCruft** [comment]
thank you logic! 

I will test it as soon as I'm home and keep updated.

**Comment by DashCruft** [comment]
Okay as it seems, you don't catch the error for when a lutris game doesn't contain a banner. 
My example game in lutris lacks a banner and this error pops up.

image

**Comment by DashCruft** [comment]
Also, even when I set a banner for the lutris game, it seems the banner doesn't load quite right.

image

**Comment by LogicApples** [comment]
> Okay as it seems, you don't catch the error for when a lutris game doesn't contain a banner. 
> My example game in lutris lacks a banner and this error pops up.
> 
> image
> 
> 
> 

oh totally forgor 💀. Will do that and also try custom banners

**Review by LegItMate — COMMENTED** [review]
And the launcher doesn't seem to work offline(confirmed on windows)

File: src/js/launchers/Minecraft.js [hunk 1]
```diff
@@ -74,19 +72,36 @@ async function getMinecraftLauncherOnLinux() {
...
+			Size: fs.statSync(Location).size,
+			Location,
+			Executable,
+			Args: [],
+		};
+	} else if (process.platform === 'linux') {
+		const homedir = require('os').userInfo().homedir;
+		const isLunarInstalled = fs.existsSync(`${homedir}/.config/lunarclient`);
+		const Location = `${homedir}/.lunarclient`;
+		const Executable = 'Lunar Client-2.11.2.AppImage';
```
[the remark this answers is not in the document]
Comment (LogicApples) [reply src/js/launchers/Minecraft.js:93]: sir thats the path lunar makes. you cant change it... :skull:

File: src/js/launchers/find-games.js [hunk 2]
```diff
@@ -235,46 +240,84 @@ function addLaunch(GameID, LauncherName) {
...
+				await exec(`cd ${homeDir} && find -name 'Lunar Client-*' | grep '.AppImage'`, (error, stdout, stderr) => {
+					processString = `${stdout.replace(/[\r\n]/gm, '')}`;
+				});
+				setTimeout(() => {
+					console.log(processString);
+					res = createProcess('cd', [homeDir, '&&', `'${processString}'`], game);
+					const jsonContent = JSON.stringify({
+						'LunarAppImageLocation': processString,
+					});
+					fs.writeFile('storage/Cache/Games/LunarCache.json', jsonContent, 'utf8', (err) => {
```
[the remark this answers is not in the document]
Comment (LogicApples) [reply src/js/launchers/find-games.js:292]: true but isn't the games object fetchd everytime you reopen lazap? that would mean finding the appimage again which means 5 secons opening of lunar everytime.
unless we ditch appimage support and only support aur package.

**Comment by LogicApples** [comment]
> And the launcher doesn't seem to work offline(confirmed on windows)

i suppose it's because of banners. that can easily be fixed. will get working right away

File: src/js/launchers/find-games.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (DashCruft) [reply src/js/launchers/find-games.js:292]: ditch appimage support. 

Thanks.

[the remark this answers is not in the document]
Comment (LogicApples) [reply src/js/launchers/find-games.js:292]: fair enough. bye

[the remark this answers is not in the document]
Comment (DashCruft) [reply src/js/launchers/find-games.js:292]: Sounds good!

**Comment by DashCruft** [comment]
The bug for the banner still exists. 
Please set a default banner for lutris games when it cannot find any.

**Comment by LogicApples** [comment]
> The bug for the banner still exists. 
> Please set a default banner for lutris games when it cannot find any.

weird it works good for me

**Pull request closed without merging**